aws/launch_template_test.go

```diff
@@ -0,0 +1,189 @@
+package aws
+
+import (
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestLaunchTemplate(t *testing.T, session *session.Session, name string) {
+	svc := ec2.New(session)
+
+	param := &ec2.CreateLaunchTemplateInput{
+		LaunchTemplateName: &name,
+		LaunchTemplateData: &ec2.RequestLaunchTemplateData{
+			InstanceType: awsgo.String("t2.micro"),
+		},
+		VersionDescription: aws.String("cloud-nuke-test-v1"),
+	}
+
+	_, err := svc.CreateLaunchTemplate(param)
+	if err != nil {
+		assert.Failf(t, "Could not create test Launch template", errors.WithStackTrace(err).Error())
+	}
+
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+}
+
+func TestListLaunchTemplates(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	uniqueTestID := "cloud-nuke-test-" + util.UniqueID()
+	createTestLaunchTemplate(t, session, uniqueTestID)
+
+	// clean up after this test
+	defer nukeAllLaunchTemplates(session, []*string{&uniqueTestID})
+
+	templateNames, err := getAllLaunchTemplates(session, region, time.Now().Add(1*time.Hour*-1), config.Config{})
+	if err != nil {
+		assert.Fail(t, "Unable to fetch list of Launch Templates")
```

Comment:
I was wondering why not to use `assert.NoErrorf()` instead of `if / assert.Fail()`?

Reply:
Thank you! Newbie golang dev here. `assert.NoError` is much better.